### PR TITLE
Add display mode for custom parameters

### PR DESCRIPTION
**Pull request by zutshiy**
## Summary

I saw an interesting issue so wanted to help out with this PR.

Adding the ability to set display modes for custom parameters.

A new parameter named `<customParametersDisplayMode>` which can be used to decide where the custom parameters are displayed. Currently it supports the below two modes:

* `START_PAGE`: Display only on the starting page (uses the `startPage` property).
* `ALL_PAGES`: Display on all the pages in the report.

**Note:**
If the property is left empty, it defaults to `START_PAGE`. So the props are no longer exclusive to the `All Scenarios Page`. They depend on whichever property `startPage` points to (default being `ALL_SCENARIOS`).

There were a bunch of changes needed so please let me know if you have any changes/enhancements you need me to work on.
I have, for now, extracted the custom parameters block to a separate `.ftl` file.

### Unit Tests
Existing unit test were updated to allow for all the changes. I don't think any new JUnits are needed, if they are please let me know.

### Closing issues
 #269 

### Documentation
Documentation as well as the `example-project` pom (with commented property) have been updated to reflect the new changes.

### CODE REVIEW

**Comment by gdonati78** [comment]
Hi @zutshiy, thanks for the PR and sorry for the late feedback. I will follow up on it Monday next week, right now the whole trivago team is on holiday.

File: plugin-code/src/test/java/com/trivago/cluecumber/rendering/pages/pojos/pagecollections/PojoTest.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
 import java.util.List;
 
 public class PojoTest {
-    private static final int EXPECTED_CLASS_COUNT = 11;
+    private static final int EXPECTED_CLASS_COUNT = 12;
```
Comment (gdonati78) [inline plugin-code/src/test/java/com/trivago/cluecumber/rendering/pages/pojos/pagecollections/PojoTest.java:17]: I checked out your fork and this test expects 11, what's the result for you?

**Comment by gdonati78** [comment]
I tried to generate reports with this, from an existing project where we use Cluecumber, but I get 
`[WARNING] ClassRealm[plugin>com.trivago.rta:cluecumber-report-plugin:2.8.0, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@2c13da15]
com.google.inject.CreationException: Unable to create injector, see the following errors:
1) No implementation for com.trivago.cluecumber.rendering.pages.templates.TemplateEngine annotated with interface org.eclipse.sisu.inject.TypeArguments$Implicit was bound.
  Did you mean?
    com.trivago.cluecumber.rendering.pages.templates.TemplateEngine bound  at org.eclipse.sisu.wire.LocatorWiring
  at org.eclipse.sisu.wire.LocatorWiring
1 error
    at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist (Errors.java:543)
    at com.google.inject.internal.InternalInjectorCreator.initializeStatically (InternalInjectorCreator.java:159)
    at com.google.inject.internal.InternalInjectorCreator.build (InternalInjectorCreator.java:106)
    at com.google.inject.Guice.createInjector (Guice.java:87)
    at com.google.inject.Guice.createInjector (Guice.java:69)
    at com.google.inject.Guice.createInjector (Guice.java:59)
    at org.codehaus.plexus.DefaultPlexusContainer.addPlexusInjector (DefaultPlexusContainer.java:481)
    at org.codehaus.plexus.DefaultPlexusContainer.discoverComponents (DefaultPlexusContainer.java:460)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.discoverPluginComponents (DefaultMavenPluginManager.java:436)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.createPluginRealm (DefaultMavenPluginManager.java:415)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.setupPluginRealm (DefaultMavenPluginManager.java:374)
    at org.apache.maven.plugin.DefaultBuildPluginManager.getPluginRealm (DefaultBuildPluginManager.java:234)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:105)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)`
    
    Did you manage to get reports generated, with custom parameters, without issues?

**Comment by zutshiy** [comment]
> @2

I did run it with the example project and am able to get the custom parameters without any issues. With `ALL_PAGES`as well as `START_PAGE`. Did not try it with a personal project though. Are you having issues running the example project with the new configuration as well (I have it commented in the pom by default) ?

File: plugin-code/src/test/java/com/trivago/cluecumber/rendering/pages/pojos/pagecollections/PojoTest.java (hunk 1 above)
Comment (zutshiy) [reply plugin-code/src/test/java/com/trivago/cluecumber/rendering/pages/pojos/pagecollections/PojoTest.java:17]: My bad. I remember updating this when I added a new class but must have forgotten to revert it later on. Fixed now.

**Comment by gdonati78** [comment]
> @gdonati78
> 
> I did run it with the example project and am able to get the custom parameters without any issues. With `ALL_PAGES`as well as `START_PAGE`. Did not try it with a personal project though. Are you having issues running the example project with the new configuration as well (I have it commented in the pom by default) ?

OK then I have to test a bit more and understand the issue, on it now

**Comment by zutshiy** [comment]
> > @gdonati78
> > I did run it with the example project and am able to get the custom parameters without any issues. With `ALL_PAGES`as well as `START_PAGE`. Did not try it with a personal project though. Are you having issues running the example project with the new configuration as well (I have it commented in the pom by default) ?
> 
> OK then I have to test a bit more and understand the issue, on it now

Ofcourse.

**Comment by gdonati78** [comment]
> > > @gdonati78
> > > I did run it with the example project and am able to get the custom parameters without any issues. With `ALL_PAGES`as well as `START_PAGE`. Did not try it with a personal project though. Are you having issues running the example project with the new configuration as well (I have it commented in the pom by default) ?
> > 
> > 
> > OK then I have to test a bit more and understand the issue, on it now
> 
> Ofcourse.

I still face the same issue in the example project as in one of our projects. I just built as new version (2.8.0) cluecumber from your fork and installed in my local maven repo, then updated the example project pom to use 2.8.0 and on `mvn clean verify` I get the same error as previously mentioned:  No implementation for com.trivago.cluecumber.rendering.pages.templates.TemplateEngine annotated with interface org.eclipse.sisu.inject.TypeArguments$Implicit was bound

I took over the project just now and need some more digging into it to understand, sorry about that, but if you have any hint or idea about what's going on please share

**Comment by bischoffdev** [comment]
Is this one still valid?

**Comment by zutshiy** [comment]
> Is this one still valid?

The build was working fine on my local and the feature is working as expected. Was waiting on @gdonati78 to verify. Perhaps you could test it on your local and let us know?

**Comment by bischoffdev** [comment]
From the code it looks really good. Will test it locally now.

**Comment by bischoffdev** [comment]
I merged it but it needs some cleanup which I am doing now.

**Comment by zutshiy** [comment]
Awesome!

**Pull request merged**